src/js/entry.js

```diff
@@ -208,3 +209,33 @@ export function rootApp() {
         );
     }
 }
+
+export function noAccess() {
+    const { store } = spinUpStore();
+    window.insights.chrome.auth.getUser().then(({ entitlements }) => {
+        // rhel has different entitlements key and URL partial
+        entitlements.rhel = entitlements.smart_management;
+        const path = location.pathname.split('/');
+        const apps = [];
+
+        /* eslint-disable camelcase */
+        const grantAccess = Object.entries(entitlements).filter(([app, { is_entitled }]) => {
+            // check if app key from entitlements is anywhere in URL and if so check if user is entitled for such app
+            apps.push(app);
+            return path.includes(app) && is_entitled;
+        });
+        /* eslint-enable camelcase */
+
+        // also grant access to other pages like settings/general
+        const untrackedApp = path.filter(value => apps.includes(value)).length < 1 ? true : false;
```

Comment:
It is generaly better to use `some` to check if one array contains something from another one
```JS
const isTrackedApp = path.some(value => app.includes(value));

if (!(grantAccess && grantAccess.length > 0) && isTrackedApp) {
```

The folter above can be changed to use `some` as well, but let's keep it there.